alert-common/src/main/java/com/synopsys/integration/alert/common/message/model/ComponentItem.java

```diff
@@ -140,7 +140,7 @@ public boolean collapseOnCategory() {
         return getCollapseOnCategory();
     }
 
-    public LinkedHashSet<LinkableItem> getComponentAttributes() {
+    public Set<LinkableItem> getComponentAttributes() {
```

Comment:
Is there a way we can keep the explicit declaration that this set is an ordered set (not necessarily sorted, but ordered)?

Reply:
Where we create the Set we are creating a LinkedHashSet. SonarCloud reasonably points out that User's should not worry about the exact implementation of the Set so just return the generic Set type.

Reply:
My worry is that it is imperative to the consumer of this object that order be maintained. If someone consumes this object by streaming over it and collecting to a generic set, then the order is lost. The API must somehow explicitly declare that this is ordered.

Reply:
Alright, thats not a bad criticism, I can change this back and ignore the rule violation.